src/main/java/mekanism/common/tile/TileEntityDimensionalStabilizer.java

```diff
@@ -0,0 +1,211 @@
+package mekanism.common.tile;
+
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import mekanism.api.Action;
+import mekanism.api.AutomationType;
+import mekanism.api.IContentsListener;
+import mekanism.api.NBTConstants;
+import mekanism.api.RelativeSide;
+import mekanism.api.math.FloatingLong;
+import mekanism.common.capabilities.energy.item.FixedUsageEnergyContainer;
+import mekanism.common.capabilities.holder.energy.EnergyContainerHelper;
+import mekanism.common.capabilities.holder.energy.IEnergyContainerHolder;
+import mekanism.common.capabilities.holder.slot.IInventorySlotHolder;
+import mekanism.common.capabilities.holder.slot.InventorySlotHelper;
+import mekanism.common.config.MekanismConfig;
+import mekanism.common.integration.computer.SpecialComputerMethodWrapper.ComputerIInventorySlotWrapper;
+import mekanism.common.integration.computer.annotation.WrappingComputerMethod;
+import mekanism.common.inventory.container.MekanismContainer;
+import mekanism.common.inventory.slot.EnergyInventorySlot;
+import mekanism.common.lib.chunkloading.IChunkLoader;
+import mekanism.common.registries.MekanismBlocks;
+import mekanism.common.tile.base.SubstanceType;
+import mekanism.common.tile.base.TileEntityMekanism;
+import mekanism.common.tile.component.TileComponentChunkLoader;
+import mekanism.common.util.MekanismUtils;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.SectionPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.block.state.BlockState;
+
+public class TileEntityDimensionalStabilizer extends TileEntityMekanism implements IChunkLoader {
+
+    public static final int MAX_LOAD_RADIUS = 2;
+    public static final int MAX_LOAD_DIAMETER = 2 * MAX_LOAD_RADIUS + 1;
+
+    protected final ChunkLoader chunkLoaderComponent;
+    protected final boolean[][] loadingChunks;
+
+    protected boolean chunkloading = false;
+
+    protected FixedUsageEnergyContainer<TileEntityDimensionalStabilizer> energyContainer;
+    @WrappingComputerMethod(wrapper = ComputerIInventorySlotWrapper.class, methodNames = "getEnergyItem")
+    private EnergyInventorySlot energySlot;
+
+    public TileEntityDimensionalStabilizer(BlockPos pos, BlockState state) {
+        super(MekanismBlocks.DIMENSIONAL_STABILIZER, pos, state);
+
+        chunkLoaderComponent = new ChunkLoader(this);
+        loadingChunks = new boolean[MAX_LOAD_DIAMETER][MAX_LOAD_DIAMETER];
+        loadingChunks[MAX_LOAD_RADIUS][MAX_LOAD_RADIUS] = true;
+    }
+
+    public boolean isChunkloadingAt(int x, int z) {
+        return loadingChunks[x][z];
+    }
+
+    public void toggleChunkloadingAt(int x, int z) {
+        loadingChunks[x][z] = !loadingChunks[x][z];
```

Comment:
Given this method is called via a packet, we should validate the `x` and `z` values are actually within the proper ranges so we don't let modified clients crash the server with an IOBE by sending invalid values,